manager/dedup.go

```diff
@@ -220,26 +221,32 @@ func (d *DedupManager) UpdateDeps(t *template.Template, deps []dep.Dependency) e
 		}
 	}
 
-	// Encode via GOB and LZW compress
-	var buf bytes.Buffer
-	compress := lzw.NewWriter(&buf, lzw.LSB, 8)
-	enc := gob.NewEncoder(compress)
-	if err := enc.Encode(&td); err != nil {
-		return fmt.Errorf("encode failed: %v", err)
+	// Compute stable hash of the data. Note we don't compute this over the actual
+	// encoded value since gob encoding does not guarantee stable ordering for
+	// maps so spuriously returns a different hash most times. See
+	// https://github.com/hashicorp/consul-template/issues/1099.
+	hash, err := hashstructure.Hash(td, nil)
+	if err != nil {
```

Comment:
not a blocker, but curious under what sort of inputs does hashstructure even ever return an error

Reply:
mostly stuff like IO failures to write value to buffer and calculate hash over it but also if any reflect issues come across types it doesn't know how to handle etc. e.g. `fmt.Errorf("unknown kind to hash: %s", k)` (https://github.com/mitchellh/hashstructure/blob/master/hashstructure.go)